src/app/components/battle.ts

```diff
@@ -0,0 +1,69 @@
+import { createElement } from '../utils/createElement'
+import { Audio } from '../audio'
+import { Game } from '../game'
+
+export class Battle {
+  private audio: Audio
+  private game: Game
+  private mainArea: HTMLElement = document.querySelector(
+    '.main-area'
+  ) as HTMLElement
+  private infoHeading: HTMLElement = document.querySelector(
+    '.info__heading'
+  ) as HTMLElement
+  private progress: HTMLElement = document.querySelector(
+    '.progress'
+  ) as HTMLElement
+  private battleContainer: HTMLElement = createElement({
+    type: 'div',
+    classes: ['battle__container'],
+  })
+
+  constructor(audio: Audio, game: Game) {
+    this.audio = audio
+    this.game = game
+  }
+
+  private createBattle() {
+    const battleEffects = createElement({
+      type: 'img',
+      classes: ['battle__effects'],
+    })
+
+    battleEffects.setAttribute('src', '../../assets/gif/battle.gif')
+
+    this.battleContainer.appendChild(battleEffects)
+  }
+
+  private renderBattle() {
+    if (this.mainArea.childElementCount === 2) {
+      this.mainArea.appendChild(this.battleContainer)
+      this.infoHeading.style.opacity = '0'
+      this.progress.style.opacity = '0'
+      this.createBattle()
+      this.audio.changeAudioSource('../../assets/audio/battle.mp3')
+    }
+  }
+
+  private removeBattle() {
+    if (this.infoHeading.style.opacity === '0') {
```

Comment:
Can we set '==' here? It should be string but who knows what will happen in the future